crypto/bn/rsaz_exp.h

```diff
@@ -52,6 +54,29 @@ int ossl_rsaz_mod_exp_avx512_x2(BN_ULONG *res1,
                                 BN_ULONG k0_2,
                                 int factor_size);
 
+static ossl_inline void bn_select_words(BN_ULONG *r, BN_ULONG mask,
+                                        const BN_ULONG *a,
+                                        const BN_ULONG *b, size_t num)
+{
+    size_t i;
+
+    for (i = 0; i < num; i++) {
+        r[i] = constant_time_select_64(mask, a[i], b[i]);
+    }
+}
+
+static ossl_inline BN_ULONG bn_reduce_once_in_place(BN_ULONG *r,
+                                                    BN_ULONG carry,
+                                                    const BN_ULONG *m,
+                                                    void *tmp, size_t num)
+{
+    BN_ULONG *tmpbn = tmp;
```

Comment:
Using `void*` and then casting to `BN_ULONG*` instead the function seems likely to cause strict aliasing violations later on. Seems better to use `BN_ULONG*` so the type system will actually notice mistakes.

Reply:
As this is just a temporary storage area for this function and we do not care about the contents after the return this should not be a problem?

Reply:
Even as temporary storage, It is UB in C to access an object with the wrong pointer type. Some violations may be more or less likely to be mangled by the compiler, but I think it's safest to not tempt fate.

Reply:
OK